internal/clients/manager.go

```diff
@@ -173,6 +184,23 @@ func (c *AdminAPIClientsManager) GatewayClients() []*adminapi.Client {
 	return lo.Values(c.readyGatewayClients)
 }
 
+func (c *AdminAPIClientsManager) GatewayClientsToConfigure() []*adminapi.Client {
+	c.lock.RLock()
+	defer c.lock.RUnlock()
+	readyGatewayClients := lo.Values(c.readyGatewayClients)
+	// With dbless mode, we should send configuration to ALL Kong gateway instances.
+	if dataplaneutil.IsDBLessMode(c.dbMode) {
+		return readyGatewayClients
+	}
+	// When Kong gateway is DB backed, we return a random admin API client
+	// since KIC only need to send requests to one instance.
+	// If there are no ready gateway clients, we return an empty list.
```

Comment:
```suggestion
	// When a gateway is DB-backed, we return a random client
	// since KIC only needs to send requests to one instance.
	// If there are no ready gateway clients, we return an empty list.
```